sdk/core/azure-core/src/main/java/com/azure/core/util/polling/SimpleSyncPoller.java

```diff
@@ -0,0 +1,284 @@
+            syncPollResponse.getValue(),
+            syncPollResponse.getRetryAfter());
+    }
+
+    private boolean matchStatus(AsyncPollResponse<T, U> currentPollResponse,
+                                LongRunningOperationStatus statusToWaitFor) {
+        if (currentPollResponse == null || statusToWaitFor == null) {
+            return false;
+        }
+        if (statusToWaitFor == currentPollResponse.getStatus()) {
+            return true;
+        }
+        return false;
+    }
+
+    private SyncPollResponse<T, U> pollingLoopSync(PollingContext<T> pollingContext) {
+        return this.pollingLoopSync(pollingContext, Optional.empty(), Optional.empty());
+    }
+
+    private SyncPollResponse<T, U> pollingLoopSync(PollingContext<T> pollingContext, LongRunningOperationStatus statusToWaitFor) {
+        return this.pollingLoopSync(pollingContext, Optional.empty(), Optional.of(statusToWaitFor));
+    }
+
+    private SyncPollResponse<T, U> pollingLoopSync(PollingContext<T> pollingContext, Duration timeout,
+                                                   LongRunningOperationStatus statusToWaitFor) {
+        return this.pollingLoopSync(pollingContext, Optional.of(timeout), Optional.of(statusToWaitFor));
+    }
+
+    private SyncPollResponse<T, U> pollingLoopSync(PollingContext<T> pollingContext, Duration timeout) {
+        return this.pollingLoopSync(pollingContext, Optional.of(timeout), Optional.empty());
+    }
+
+    private SyncPollResponse<T, U> pollingLoopSync(PollingContext<T> pollingContext, Optional<Duration> timeout,
+                                                   Optional<LongRunningOperationStatus> statusToWaitFor) {
+        boolean timeBound = timeout.isPresent();
+        long startTime = System.currentTimeMillis();
+        PollResponse<T> pollResponse = pollOperation.apply(pollingContext);
+        pollingContext.setLatestResponse(pollResponse);
+        SyncPollResponse<T, U> intermediatePollResponse = new SyncPollResponse<>(pollingContext, this.cancelOperation, this.fetchResultOperation);
+        while (!pollResponse.getStatus().isComplete()
+            && (timeBound ? (System.currentTimeMillis() - startTime) < timeout.get().toMillis() : true)) {
+            try {
+                if (statusToWaitFor.isPresent() && pollResponse.getStatus().equals(statusToWaitFor.get())) {
+                    return intermediatePollResponse;
+                }
+                Thread.sleep(getDelay(pollResponse).toMillis());
+                // Document that Poll operation respects timeout, cannot interrupt it from here.
+                pollResponse = pollOperation.apply(pollingContext);
+                pollingContext.setLatestResponse(pollResponse);
+                intermediatePollResponse = new SyncPollResponse<>(pollingContext, this.cancelOperation, this.fetchResultOperation);
+            } catch (InterruptedException ex) {
+                throw LOGGER.logExceptionAsError(Exceptions.propagate(ex));
```

Comment:
Does this need to use `Exceptions.propagate` given this shouldn't be called from an asynchronous code path